### PR TITLE
kv: Remove old kvclient related code.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

This PR try to remove old kv client code, included unnecessary test and benchmark code.

### What is changed and how it works?

* remove old kv client code
* remove related code in test and benchmark.

### Check List 

Tests 

 - No code

Code changes

Side effects
Related changes

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Comment by 3AceShowHand** [comment]
This PR needs to be cherry-picked to `release-4.0` and `release-5.0`, but should be merged after new kv client enabled

File: cdc/kv/client.go [hunk 1]
```diff
@@ -1123,6 +1021,13 @@ func (s *eventFeedSession) getRPCContextForRegion(ctx context.Context, id tikv.R
 	return rpcCtx, nil
 }
 
+// receiveFromStreamV2 receives gRPC messages from a stream continuously and sends
```
Comment (amyangfei) [inline cdc/kv/client.go:1024]:
```suggestion
// receiveFromStreamV receives gRPC messages from a stream continuously and sends
```

**Review by amyangfei — COMMENTED** [review]
rest lgtm

**Review by amyangfei — APPROVED** [review]
Good job

File: cdc/kv/client_test.go [hunk 2]
```diff
@@ -1586,13 +1576,6 @@ func (s *etcdSuite) TestStreamRecvWithErrorIOEOF(c *check.C) {
 
 	// test client v2
```
Comment (overvenus) [inline cdc/kv/client_test.go:1577]:
```suggestion
```

**Review by overvenus — APPROVED** [review]
LGTM

**Comment by lonng** [comment]
Please refine the PR description and remove unnecessary checklist items.

**Comment by 3AceShowHand** [comment]
@amyangfei 

I want to make it clear that shall we cherry-pick this to release-4.0 / release-5.0? Eventhough the robot didn't do it.

**Pull request merged**